### PR TITLE
Force species in tests

### PR DESCRIPTION
**Pull request by AlexOn1ine**
This forces all species to be always enabled in tests. Prevent users from having a lot of tests failing. Especially applies to tests with forms because there are no assumes for them.

### CODE REVIEW

File: include/config/test.h [hunk 1]
```diff
@@ -10,4 +10,1125 @@
...
+#define P_GEN_3_CROSS_EVOS               TRUE
+#undef P_GEN_4_CROSS_EVOS
+#define P_GEN_4_CROSS_EVOS               TRUE
+#undef P_GEN_6_CROSS_EVOS
+#define P_GEN_6_CROSS_EVOS               TRUE
+#undef P_GEN_8_CROSS_EVOS
+#define P_GEN_8_CROSS_EVOS               TRUE
+#undef P_GEN_9_CROSS_EVOS
+#define P_GEN_9_CROSS_EVOS               TRUE
+#undef P_FAMILY_BULBASAUR
```
Comment (AsparagusEduardo) [inline include/config/test.h:55]:
```suggestion
#define P_GEN_9_CROSS_EVOS               TRUE

#undef P_FAMILY_BULBASAUR
```

File: include/config/test.h [hunk 2]
```diff
@@ -10,4 +10,1125 @@
...
+#define P_ULTRA_BURST_FORMS              TRUE
+#undef P_GIGANTAMAX_FORMS
+#define P_GIGANTAMAX_FORMS               TRUE
+#undef P_TERA_FORMS
+#define P_TERA_FORMS                     TRUE
+#undef P_FUSION_FORMS
+#define P_FUSION_FORMS                   TRUE
+#undef P_REGIONAL_FORMS
+#define P_REGIONAL_FORMS                 TRUE
+#undef P_ALOLAN_FORMS
```
Comment (AsparagusEduardo) [inline include/config/test.h:27]: we don't need the ones for stuff like `P_REGIONAL_FORMS` if they're only used as an easy way to disable multiple other defines
```suggestion
#define P_FUSION_FORMS                   TRUE
#undef P_ALOLAN_FORMS
```

File: include/config/test.h [hunk 3]
```diff
@@ -10,4 +10,1125 @@
...
+#define P_PALDEAN_FORMS                  TRUE
+#undef P_PIKACHU_EXTRA_FORMS
+#define P_PIKACHU_EXTRA_FORMS            TRUE
+#undef P_COSPLAY_PIKACHU_FORMS
+#define P_COSPLAY_PIKACHU_FORMS          TRUE
+#undef P_CAP_PIKACHU_FORMS
+#define P_CAP_PIKACHU_FORMS              TRUE
+#undef P_CROSS_GENERATION_EVOS
+#define P_CROSS_GENERATION_EVOS          TRUE
+#undef P_GEN_2_CROSS_EVOS
```
Comment (AsparagusEduardo) [inline include/config/test.h:43]:
```suggestion
#define P_CAP_PIKACHU_FORMS              TRUE
#undef P_GEN_2_CROSS_EVOS
```

File: include/config/test.h [hunk 4]
```diff
@@ -10,4 +10,1125 @@
...
+#define P_FAMILY_ZAPDOS                  TRUE
+#undef P_FAMILY_MOLTRES
+#define P_FAMILY_MOLTRES                 TRUE
+#undef P_FAMILY_DRATINI
+#define P_FAMILY_DRATINI                 TRUE
+#undef P_FAMILY_MEWTWO
+#define P_FAMILY_MEWTWO                  TRUE
+#undef P_FAMILY_MEW
+#define P_FAMILY_MEW                     TRUE
+#undef P_FAMILY_CHIKORITA
```
Comment (AsparagusEduardo) [inline include/config/test.h:209]:
```suggestion
#define P_FAMILY_MEW                     TRUE

#undef P_FAMILY_CHIKORITA
```

File: include/config/test.h [hunk 5]
```diff
@@ -10,4 +10,1125 @@
...
+#define P_FAMILY_SUICUNE                 TRUE
+#undef P_FAMILY_LARVITAR
+#define P_FAMILY_LARVITAR                TRUE
+#undef P_FAMILY_LUGIA
+#define P_FAMILY_LUGIA                   TRUE
+#undef P_FAMILY_HO_OH
+#define P_FAMILY_HO_OH                   TRUE
+#undef P_FAMILY_CELEBI
+#define P_FAMILY_CELEBI                  TRUE
+#undef P_FAMILY_TREECKO
```
Comment (AsparagusEduardo) [inline include/config/test.h:311]:
```suggestion
#define P_FAMILY_CELEBI                  TRUE

#undef P_FAMILY_TREECKO
```

File: include/config/test.h [hunk 6]
```diff
@@ -10,4 +10,1125 @@
...
+#define P_FAMILY_KYOGRE                  TRUE
+#undef P_FAMILY_GROUDON
+#define P_FAMILY_GROUDON                 TRUE
+#undef P_FAMILY_RAYQUAZA
+#define P_FAMILY_RAYQUAZA                TRUE
+#undef P_FAMILY_JIRACHI
+#define P_FAMILY_JIRACHI                 TRUE
+#undef P_FAMILY_DEOXYS
+#define P_FAMILY_DEOXYS                  TRUE
+#undef P_FAMILY_TURTWIG
```
Comment (AsparagusEduardo) [inline include/config/test.h:455]:
```suggestion
#define P_FAMILY_DEOXYS                  TRUE

#undef P_FAMILY_TURTWIG
```

File: include/config/test.h [hunk 7]
```diff
@@ -10,4 +10,1125 @@
...
+#define P_FAMILY_LANDORUS                TRUE
+#undef P_FAMILY_KYUREM
+#define P_FAMILY_KYUREM                  TRUE
+#undef P_FAMILY_KELDEO
+#define P_FAMILY_KELDEO                  TRUE
+#undef P_FAMILY_MELOETTA
+#define P_FAMILY_MELOETTA                TRUE
+#undef P_FAMILY_GENESECT
+#define P_FAMILY_GENESECT                TRUE
+#undef P_FAMILY_CHESPIN
```
Comment (AsparagusEduardo) [inline include/config/test.h:707]:
```suggestion
#define P_FAMILY_GENESECT                TRUE

#undef P_FAMILY_CHESPIN
```

File: include/config/test.h [hunk 8]
```diff
@@ -10,4 +10,1125 @@
...
+#define P_FAMILY_YVELTAL                 TRUE
+#undef P_FAMILY_ZYGARDE
+#define P_FAMILY_ZYGARDE                 TRUE
+#undef P_FAMILY_DIANCIE
+#define P_FAMILY_DIANCIE                 TRUE
+#undef P_FAMILY_HOOPA
+#define P_FAMILY_HOOPA                   TRUE
+#undef P_FAMILY_VOLCANION
+#define P_FAMILY_VOLCANION               TRUE
+#undef P_FAMILY_ROWLET
```
Comment (AsparagusEduardo) [inline include/config/test.h:781]:
```suggestion
#define P_FAMILY_VOLCANION               TRUE

#undef P_FAMILY_ROWLET
```

File: include/config/test.h [hunk 9]
```diff
@@ -10,4 +10,1125 @@
...
+#define P_FAMILY_POIPOLE                 TRUE
+#undef P_FAMILY_STAKATAKA
+#define P_FAMILY_STAKATAKA               TRUE
+#undef P_FAMILY_BLACEPHALON
+#define P_FAMILY_BLACEPHALON             TRUE
+#undef P_FAMILY_ZERAORA
+#define P_FAMILY_ZERAORA                 TRUE
+#undef P_FAMILY_MELTAN
+#define P_FAMILY_MELTAN                  TRUE
+#undef P_FAMILY_GROOKEY
```
Comment (AsparagusEduardo) [inline include/config/test.h:891]:
```suggestion
#define P_FAMILY_MELTAN                  TRUE

#undef P_FAMILY_GROOKEY
```

File: include/config/test.h [hunk 10]
```diff
@@ -10,4 +10,1125 @@
...
+#define P_FAMILY_REGIDRAGO               TRUE
+#undef P_FAMILY_GLASTRIER
+#define P_FAMILY_GLASTRIER               TRUE
+#undef P_FAMILY_SPECTRIER
+#define P_FAMILY_SPECTRIER               TRUE
+#undef P_FAMILY_CALYREX
+#define P_FAMILY_CALYREX                 TRUE
+#undef P_FAMILY_ENAMORUS
+#define P_FAMILY_ENAMORUS                TRUE
+#undef P_FAMILY_SPRIGATITO
```
Comment (AsparagusEduardo) [inline include/config/test.h:987]:
```suggestion
#define P_FAMILY_ENAMORUS                TRUE

#undef P_FAMILY_SPRIGATITO
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Comment by Bassoonian** [comment]
We can consider this out of scope, but I really wonder if a `TEST_OVERRIDE` or `OVERRIDE` macro would be easier here so we don't need to manually undefine it and then define it once again

**Comment by AlexOn1ine** [comment]
> We can consider this out of scope, but I really wonder if a `TEST_OVERRIDE` or `OVERRIDE` macro would be easier here so we don't need to manually undefine it and then define it once again

Could you elaborate what you mean?

**Comment by AlexOn1ine** [comment]
> We can consider this out of scope, but I really wonder if a `TEST_OVERRIDE` or `OVERRIDE` macro would be easier here so we don't need to manually undefine it and then define it once again

Can we do this in an other pr? I don't want to modify this right now

**Comment by Bassoonian** [comment]
Sure, fine by me! Feel free to open an issue and I'll look into it later

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**